toolkit/tools/imagegen/configuration/systemconfig.go

```diff
@@ -70,6 +70,9 @@ func (s *SystemConfig) IsValid() (err error) {
 	// IsDefault must be validated by a parent struct
 
 	// Validate BootType
+	// if (s.BootType != "efi" && s.BootType != "legacy") {
```

Comment:
Is it to uncomment one day? Or just forgot to remove?

Reply:
Ahh, thank you! This was from when I was trying to test config validation. It didn't seem appropriate for this PR though. I will remove it, thanks Betty!

Reply:
Could you also add a test case for these in either `systemconfig_test` or `imageconfigvalidator_test` (You can check the test coverage with `sudo make go-test-coverage`, it spits out a nice html file with everything).

Reply:
This seems like a legitimate thing to validate, if you think there is value feel free to leave it in place. Ideally our whole schema should be validated (its a long ways to go still though).

Reply:
How should we handle the situation where BootType is not set in the imageconfig? As the validation is currently "written", it will throw a fit if BootType is not set and I imagine this will break a lot of people's imageconfigs and upset a bunch of people

Reply:
Talked offline, "" is a valid config in many situations. Maybe add some logic to check what those situations are. (maybe check if the install target is set for example?)

Reply:
Validation added, and BootType testcase added as well!